src/extractor/guidance/roundabout_handler.cpp

```diff
@@ -131,43 +132,59 @@ void RoundaboutHandler::invalidateExitAgainstDirection(const NodeID from_nid,
     // *     | \░        | \          |    *
     // * <--'   \    <--'   v     <--'     *
     // *************************************
-    bool roundabout_entry_first = false;
-    auto invalidate_from = intersection.end(), invalidate_to = intersection.end();
-    for (auto road = intersection.begin(); road != intersection.end(); ++road)
+    const auto size = intersection.size();
+    auto intersection_entry = size, intersection_exit = size, road = size;
+    for (std::size_t index = 0; index < size; ++index)
     {
-        const auto &edge_data = node_based_graph.GetEdgeData(road->eid);
+        const auto &edge_data = node_based_graph.GetEdgeData(intersection[index].eid);
         if (edge_data.roundabout || edge_data.circular)
         {
             if (edge_data.reversed)
-            {
-                if (roundabout_entry_first)
-                { // invalidate turns in range exit..end
-                    invalidate_from = road + 1;
-                    invalidate_to = intersection.end();
-                }
-                else
-                { // invalidate turns in range begin..exit
-                    invalidate_from = intersection.begin() + 1;
-                    invalidate_to = road;
-                }
-            }
+                intersection_exit = index;
             else
-            {
-                roundabout_entry_first = true;
-            }
+                intersection_entry = index;
+        }
+        else if (road == size || intersection[road].angle == intersection[index].angle)
```

Comment:
I am unsure whether I understand this correctly. You are choosing the last road of a series of roads with same angles that is not a roundabout?

So on `(0:true), (0:false), (0:false)`, which one would you end up choosing (angle : is_roundabout). Shouldn't you compare to the `via_eid` somehow? Or is this mainly about not crashing?

Wouldn't the order of these roads be more or less random?

I feel that for a correct fix for the overarching problem, we would have to do some tie-breaking mechanisms in the road generation to ensure that these angles are always sorted correctly.